ambry-cloud/src/main/java/com/github/ambry/cloud/azure/CosmosDataAccessor.java

```diff
@@ -559,25 +608,89 @@ public long deprecateContainers(Set<ContainerDeletionEntry> deprecatedContainers
+   * @throws DocumentClientException in case of any error.
    */
-  public Set<ContainerDeletionEntry> getDeprecatedContainers(int maxEntries) {
+  public Set<CosmosContainerDeletionEntry> getDeprecatedContainers(int maxEntries) throws DocumentClientException {
     String query = String.format(DEPRECATED_CONTAINERS_QUERY, maxEntries);
     Timer timer = new Timer();
-    Iterator<FeedResponse<Document>> iterator =
-        executeCosmosQuery(cosmosDeletedContainerCollectionLink, null, new SqlQuerySpec(query), new FeedOptions(),
-            timer).getIterator();
-    Set<ContainerDeletionEntry> containerDeletionEntries = new HashSet<>();
-    while (iterator.hasNext()) {
-      FeedResponse<Document> response = iterator.next();
-      response.getResults()
-          .iterator()
-          .forEachRemaining(
-              doc -> containerDeletionEntries.add(ContainerDeletionEntry.fromJson(new JSONObject(doc.toJson()))));
+    Set<CosmosContainerDeletionEntry> containerDeletionEntries = new HashSet<>();
+    try {
+      Iterator<FeedResponse<Document>> iterator =
+          executeCosmosQuery(cosmosDeletedContainerCollectionLink, null, new SqlQuerySpec(query), new FeedOptions(),
+              timer).getIterator();
+      while (iterator.hasNext()) {
+        FeedResponse<Document> response = iterator.next();
+        response.getResults()
+            .iterator()
+            .forEachRemaining(doc -> containerDeletionEntries.add(
+                CosmosContainerDeletionEntry.fromJson(new JSONObject(doc.toJson()))));
+      }
+    } catch (RuntimeException rex) {
+      if (rex.getCause() instanceof DocumentClientException) {
+        logger.warn("Get deprecated containers query {} got {}", query,
+            ((DocumentClientException) rex.getCause()).getStatusCode());
+        throw (DocumentClientException) rex.getCause();
+      }
+      throw rex;
     }
     return containerDeletionEntries;
   }
 
+  /**
+   * Update the container deletion entry document in the CosmosDB collection.
+   * @param containerId the container id for which document is replaced.
+   * @param accountId the account id for which document is replaced.
+   * @param updateFields {@link BiConsumer} object to use as callback to update the required fields.
+   * @return the {@link ResourceResponse} returned by the operation, if successful.
+   * Returns {@Null} if the field already has the specified value.
+   * @throws DocumentClientException if the record was not found or if the operation failed.
+   */
+  ResourceResponse<Document> updateContainerDeletionEntry(short containerId, short accountId,
+      BiConsumer<Document, AtomicBoolean> updateFields) throws DocumentClientException {
+
+    // Read the existing record
+    String id = CosmosContainerDeletionEntry.generateContainerDeletionEntryId(accountId, containerId);
+    String docLink = getContainerDeletionEntryDocumentLink(id);
+    RequestOptions options = getRequestOptions(id);
```

Comment:
A lot of this logic is very similar to updateMetadata().  Possible to reuse/combine?